File: android/entities/src/main/java/com/tiagohs/entities/contents/ContentBlockSpecial.kt
package com.tiagohs.entities.contents

import com.google.gson.annotations.SerializedName
import com.tiagohs.entities.click.Click
import com.tiagohs.entities.image.Image
import java.io.Serializable

class ContentBlockSpecial(
    @SerializedName("title")
    val title: String? = null,

    @SerializedName("description")
    val description: String,

    @SerializedName("credits")
    val credits: String? = null,

    @SerializedName("image")
    val image: Image? = null,

    @SerializedName("click")
    val click: Click? = null
): Content(), Serializable

File: ios/HistoryCinema/HistoryCinema/Entities/LocalFiles/Contents/ContentBlockSpecial.swift
//
//  ContentBlockSpecial.swift
//  HistoryCinema
//
//  Created by Tiago Henrique da Silva on 21/02/21.
//

import Foundation
import ObjectMapper

class ContentBlockSpecial: Content {
    var title: String?
    var description: String!
    var credits: String?
    var image: LocalImage?
    var click: Click?
    
    required init?(map: Map) {
        super.init(map: map)
    }

    override func mapping(map: Map) {
        super.mapping(map: map)
        
        title                                               <- map["title"]
        description                                         <- map["description"]
        credits                                             <- map["credits"]
        image                                               <- map["image"]
        click                                               <- map["click"]
    }
}
